src/librustc_mir/build/expr/as_temp.rs

```diff
@@ -85,9 +85,8 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
 
         unpack!(block = this.into(&Place::Local(temp), block, expr));
 
-        // In constants, temp_lifetime is None. We should not need to drop
-        // anything because no values with a destructor can be created in
-        // a constant at this time, even if the type may need dropping.
+        // In constants, temp_lifetime is None. We do not drop anything because
```

Comment:
It's not `None` except for the outermost scope. We need to keep all other drops in, to get errors about them (since they *will* execute in the future).

Reply:
cc @nikomatsakis